src/third_party/dartino/object_memory.h

```diff
@@ -536,4 +539,23 @@ class ObjectMemory {
   static Mutex* spare_chunk_mutex_;
 };
 
+inline void write_sentinel_at(uword address) {
+  ASSERT(sizeof(Object*) == SENTINEL_SIZE);
+  *reinterpret_cast<Object**>(address) = chunk_end_sentinel();
+}
+
+uword SemiSpace::try_allocate(uword size) {
+  // Make sure there is room for chunk end sentinel by using > instead of >=.
+  // Use this ordering of the comparison to avoid very large allocations
+  // turning into 'successful' allocations of negative size.
+  if (limit_ - top_ > size) {
```

Comment:
Maybe use bailout instead? If you go this route, do remember to update the comment about using `>`.

```suggestion
  if (limit_ - top_ <= size) return 0;
```